app/src/main/java/com/beemdevelopment/aegis/ui/views/EntryAdapter.java

```diff
@@ -424,7 +429,16 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, int position)
             boolean paused = _pauseFocused && entry == _focusedEntry;
             boolean dimmed = (_highlightEntry || _tempHighlightEntry) && _focusedEntry != null && _focusedEntry != entry;
             boolean showProgress = entry.getInfo() instanceof TotpInfo && ((TotpInfo) entry.getInfo()).getPeriod() != getMostFrequentPeriod();
-            entryHolder.setData(entry, _codeGroupSize, _accountNamePosition, _showIcon, showProgress, hidden, paused, dimmed);
+            boolean showAccountName = true;
+            if (_onlyShowNecessaryAccountNames) {
+                // Only show account name when there's multiple entries found with the same issuer.
+                showAccountName = _entries.stream()
```

Comment:
This becomes expensive pretty quickly when scrolling through a list with lots of entries, but it's probably not worth trying to optimize this until we actually encounter problems with this.